sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -651,7 +651,7 @@ public void validate(PBegin input) {
         }
 
         final String executingProject = bqOptions.getProject();
-        if (query != null && !Strings.isNullOrEmpty(query.get())) {
+        if (query != null && (!query.isAccessible() || !Strings.isNullOrEmpty(query.get()))) {
```

Comment:
I'm  a little wary here -- if you look below we're using `uuid`, which is computed at apply time. But we need these to be different for every job. [BEAM-758](https://issues.apache.org/jira/browse/BEAM-758)

This seems like a problem we need to solve before ValueProvider reaches its full potential.